### PR TITLE
DB module for iOS

### LINKED ISSUE #76: Make the DB module work on iOS
## Kind (Required)

- Improvement

 ## Overview (Required)

- Add an iOS target and try the KeychainSettings or AppleSettings implementation of russhwolf/multiplatform-settings.

 ## Links	

- https://github.com/russhwolf/multiplatform-settings#apple-keychain

### PR DESCRIPTION
**Pull request by watanavex**
## Issue
- close #76

## Overview (Required)
- Add iOS target
- ~~Add UserDefaults wrapper (reference from russhwolf/multiplatform-settings)~~
- Add DataStoreUserDataStore of iOS implementation

## Links
-https://github.com/russhwolf/multiplatform-settings/blob/master/multiplatform-settings/src/appleMain/kotlin/com/russhwolf/settings/AppleSettings.kt

## Screenshot
🙅‍♂️

### CODE REVIEW

File: data/db/src/iosTest/kotlin/io/github/droidkaigi/feeder/data/DataStoreUserDataStoreTest.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package io.github.droidkaigi.feeder.data
+
+import kotlin.test.AfterTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlinx.coroutines.flow.first
+import platform.Foundation.NSUserDefaults
+
+class DataStoreUserDataStoreTest {
```
Comment (takahirom) [inline data/db/src/iosTest/kotlin/io/github/droidkaigi/feeder/data/DataStoreUserDataStoreTest.kt:9]: 🆒

File: data/db/src/iosMain/kotlin/io/github/droidkaigi/feeder/data/AppleSettings.kt [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+import kotlin.native.concurrent.freeze
+import platform.Foundation.NSNotification
+import platform.Foundation.NSNotificationCenter
+import platform.Foundation.NSUserDefaults
+import platform.Foundation.NSUserDefaultsDidChangeNotification
+import platform.darwin.NSObjectProtocol
+
+/*
+ Reference:
+ https://github.com/russhwolf/multiplatform-settings/blob/master/multiplatform-settings/src/appleMain/kotlin/com/russhwolf/settings/AppleSettings.kt
```
Comment (takahirom) [inline data/db/src/iosMain/kotlin/io/github/droidkaigi/feeder/data/AppleSettings.kt:17]: I have a question! Why did you need a copy of the code? 🙋

Comment (watanavex) [reply data/db/src/iosMain/kotlin/io/github/droidkaigi/feeder/data/AppleSettings.kt:17]: I may have made a mistake...
Thank you so much🙏
I will investigate‼️

Maybe I shouldn't have added the flag to `gradle.properties` 🤔  (#298)

Comment (takahirom) [reply data/db/src/iosMain/kotlin/io/github/droidkaigi/feeder/data/AppleSettings.kt:17]: Thank you for the investigation! Kotlin MPP requires a lot of trial and error, so let's try it 👍

Comment (watanavex) [reply data/db/src/iosMain/kotlin/io/github/droidkaigi/feeder/data/AppleSettings.kt]: I'm guessing the problem is that `multiplatform-settings` doesn't have the` kotlin.mpp.enableGranularSourceSetsMetadata` flag.
Syntax highlighting doesn't work even with #298.
I tried a workaround to add ios target. 9e2cd3a

File: data/db/build.gradle [hunk 3]
```diff
@@ -19,20 +19,14 @@ tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile.class).all {
...
-//        if (onPhone) {
-//            iosArm64("ios")
-//        } else {
-//            iosX64("ios")
-//        }
-//    } else {
-//        ios()
-//    }
+
+    // workaround: select iOS target platform depending on the Xcode environment variables
```
Comment (takahirom) [inline data/db/build.gradle:23]: Thank you for your comment!
In general, it's a good idea to write in the comments why you implemented it this way.  (from: https://google.github.io/eng-practices/review/reviewer/looking-for.html )

```kotlin
// workaround: select iOS target platform depending on the Xcode environment variables
// The multiplatform libraries do not support hierarchical structures, which breaks the IDE's completion.
// https://kotlinlang.org/docs/mpp-share-on-platforms.html#share-code-on-similar-platforms
```

**Comment by takahirom** [comment]
Support for sqldelight has been merged into the main branch. Can you merge it from the main branch into this branch and check it out? 🙇 
https://github.com/DroidKaigi/conference-app-2021/pull/275

File: data/db/build.gradle (hunk 3 above)
Comment (watanavex) [reply data/db/build.gradle:23]: Great idea 👍 
Thank you for always politely reviewing.
感謝しかありません🙏

**Comment by watanavex** [comment]
@takahirom  IOS tests are not running in the github actions workflow.
I want to fix it in another PR. May I?

**Review by takahirom — APPROVED** [review]
Thank you! We can use DataStore for Android and user defaults for iOS!

**Comment by takahirom** [comment]
> @takahirom IOS tests are not running in the github actions workflow.
I want to fix it in another PR. May I?

Thank you!!!! please fix it 🙏 🙏 🙏
